src/Tools/ILAsm/IlAsmDeploy.csproj

```diff
@@ -3,15 +3,58 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>netcoreapp2.1</TargetFramework>
-    <RuntimeIdentifiers>$(RoslynPortableRuntimeIdentifiers)</RuntimeIdentifiers>
     <OutputType>Exe</OutputType>
+
     <!-- 
       When the project is published as a self-contained app do not require the latest runtime patch. 
       This is ok since the app is only used for testing.
     -->
     <TargetLatestRuntimePatch>false</TargetLatestRuntimePatch>
+    <SelfContained>true</SelfContained>
+
+    <!-- Do not build again when publishing -->
+    <NoBuild>true</NoBuild>
+
+    <PublishDir>$(ArtifactsDir)Tools\ILAsm\</PublishDir>
   </PropertyGroup>
+
+  <Choose>
+    <When Condition="'$(OS)' == 'Windows_NT'">
+      <PropertyGroup>
+        <RuntimeIdentifier>win-x64</RuntimeIdentifier>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::OSX)))' == 'true'">
+      <PropertyGroup>
+        <RuntimeIdentifier>osx-x64</RuntimeIdentifier>
+      </PropertyGroup>
+    </When>
+    <When Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Linux)))' == 'true'">
+      <PropertyGroup>
+        <RuntimeIdentifier>linux-x64</RuntimeIdentifier>
+      </PropertyGroup>
+    </When>
+  </Choose>
+
   <ItemGroup>
     <PackageReference Include="Microsoft.NETCore.ILAsm" Version="$(MicrosoftNETCoreILAsmVersion)" />
   </ItemGroup>
+
+  <PropertyGroup>
+    <_IlasmIncrementalBuildMarker>$(PublishDir).v$(MicrosoftNETCoreILAsmVersion)</_IlasmIncrementalBuildMarker>
+  </PropertyGroup>
+
+  <Target Name="_PublishILAsm"
+          AfterTargets="Build"
+          Inputs="$(MSBuildThisFileFullPath)"
+          Outputs="$(_IlasmIncrementalBuildMarker)">
+
+    <Error Text="Unknown OS" Condition="'$(RuntimeIdentifier)' == ''" />
+
+    <MSBuild Projects="$(MSBuildThisFileFullPath)" Targets="Publish"/>
```

Comment:
Why is this better than say making the default target Publish?

Reply:
Unfortunately, Publish isn't incremental (i'm gonna file a bug on that). So we need to have a sentinel file that we use to check if we need to execute Publish again.

Reply:
https://github.com/dotnet/sdk/issues/2573